src/coreclr/pal/src/thread/context.cpp

```diff
@@ -436,7 +462,7 @@ void CONTEXTToNativeContext(CONST CONTEXT *lpContext, native_context_t *native)
 #undef ASSIGN_REG
 
 #if !HAVE_FPREGS_WITH_CW
-#if HAVE_GREGSET_T || HAVE_GREGSET_T
+#if (HAVE_GREGSET_T || HAVE_GREGSET_T) && !defined(HOST_S390X)
```

Comment:
Can you please remove the added part of the condition? It doesn't seem to be needed, if I read your change below correctly, s390x has the native->uc_mcontext.fpregs / HAVE_GREGSET_T.
The same at the other place in this file below.

Reply:
Unfortunately it is needed.  While there is indeed a native->uc_mcontext.fpregs field, it is not a pointer but rather an embedded struct on our platform, so comparing it against nullptr results in a compiler error.

Reply:
Ok, thanks for the explanation